orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/OrcaMessageHandler.kt

```diff
@@ -116,8 +116,21 @@ internal interface OrcaMessageHandler<M : Message> : MessageHandler<M> {
   fun PipelineExecution.shouldQueue(): Boolean {
     val configId = pipelineConfigId
     return when {
-      !isLimitConcurrent -> false
       configId == null -> false
+      !isLimitConcurrent -> {
+        return when {
+          getMaxConcurrentExecutions() <= 0 -> false
```

Comment:
Why less than 0?

Reply:
If `isLimitConcurrent == false && maxConcurrentExecutions <= 0` then don't queue. This implements/keeps the current logic, where `isLimitConcurrent == false` then don't queue.

Reply:
Shouldn't it just return false always, if limitConcurrent is false? Why is the check for maxConcurrent necessary?

Reply:
Shouldn't it just return false always, if limitConcurrent is false? Why is the check for maxConcurrent necessary?

Reply:
It needs to return true if `isLimitConcurrent == false && maxConcurrentExecutions > 0 && RUNNING >= maxConcurrentExecutions`. If makes more readable, I can change to:

      !isLimitConcurrent -> {
        return when {
          getMaxConcurrentExecutions() > 0 -> {
            val criteria = ExecutionCriteria().setPageSize(maxConcurrentExecutions+2).setStatuses(RUNNING)
            repository
              .retrievePipelinesForPipelineConfigId(configId, criteria)
              .filter { it.id != id }
              .count()
              .toBlocking()
              .first() >= getMaxConcurrentExecutions()
          }
          else -> false
        }
      }

Reply:
Ah my bad, I had the implication of limitConcurrent as "have an upper limit on concurrent" (which is obviously what you're implementing) rather than "prevent concurrent altogether".

Sorry!